### ISSUE TITLE
[libc++] False positive -Wthread-safety-analysis with std::lock()

### ISSUE THREAD

**Issue by rupprecht** [llvm/llvm-project#151733]
After #117497, using `std::lock` to acquire a mutex (i.e. not calling `.lock()` on the mutex itself) appears to not register that it acquires a lock, which can lead to erroneous thread safety warnings later that it needs to be held.

Live example: https://godbolt.org/z/vTrfK1f1r

```c++
void func() {
    std::mutex a, b;
    std::lock(a, b);
    // error: calling function 'lock_guard' requires holding mutex 'a' exclusively [-Werror,-Wthread-safety-analysis]
    std::lock_guard<std::mutex> a_guard(a, std::adopt_lock);
    // error: calling function 'lock_guard' requires holding mutex 'b' exclusively [-Werror,-Wthread-safety-analysis]
    std::lock_guard<std::mutex> b_guard(b, std::adopt_lock);
}
```

Real example in LLDB: https://github.com/llvm/llvm-project/blob/6c072c06cc59b56f83014450afe82f03666407ad/lldb/source/Breakpoint/BreakpointLocationCollection.cpp#L180-L185

Also, use of this pattern when needing to acquire more than one lock is suggested at https://en.cppreference.com/w/cpp/thread/lock.html#Example

Here's a suggested fix that works for the breakages I've seen:
```diff
--- a/libcxx/include/mutex
+++ b/libcxx/include/mutex
@@ -351,7 +351,7 @@
 #    endif // _LIBCPP_CXX03_LANG

 template <class _L0, class _L1>
-_LIBCPP_HIDE_FROM_ABI void lock(_L0& __l0, _L1& __l1) {
+_LIBCPP_HIDE_FROM_ABI void lock(_L0& __l0, _L1& __l1) _LIBCPP_ACQUIRE_CAPABILITY(__l0, __l1) {
   while (true) {
     {
       unique_lock<_L0> __u0(__l0);
```
